[identity] update TS guide (#25498)
Azure/azure-sdk-for-js: sdk/identity/identity/samples/AzureIdentityExamples.md

| --------------------------------------------------------------------------------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |

> |[AzureDeveloperCliCredential](#authenticatin-a-user-account-with-azure-developer-cli) | Authenticate in a development environment with Azure Developer CLI. | [Install the Azure Developer CLI](https://learn.microsoft.com/azure/developer/azure-developer-cli/install-azd) and login using the `azd auth login` [command](https://learn.microsoft.com/azure/developer/azure-developer-cli/reference?source=recommendations#azd-auth-login).

Reviewer: ```suggestion
|[AzureDeveloperCliCredential](#authenticatin-a-user-account-with-azure-developer-cli) | Authenticate in a development environment with Azure Developer CLI. | [Install the Azure Developer CLI](https://learn.microsoft.com/azure/developer/azure-developer-cli/install-azd) and login using the [`azd auth login` command](https://learn.microsoft.com/azure/developer/azure-developer-cli/reference?source=recommendations#azd-auth-login).
```

| [AzurePowerShellCredential](#authenticating-a-user-account-with-azure-powershell) | Authenticate in a development environment with Azure PowerShell.                                                                                                                                                                                                                                                       | [Install Azure PowerShell](https://docs.microsoft.com/powershell/azure/install-az-ps) and [login using the `Connect-AzAccount` cmdlet](https://docs.microsoft.com/powershell/azure/authenticate-azureps) |

> | [DefaultAzureCredential](#authenticating-with-defaultazurecredential)             | Tries `AzureDeveloperCliCredential`, `AzureCliCredential`, `AzurePowerShellCredential`, and other credentials sequentially until one of them succeeds. Use this to have your application authenticate using developer tools, service principals or managed identity based on what is available in the current environment without changing your code. |

Reviewer: ```suggestion
| [DefaultAzureCredential](#authenticating-with-defaultazurecredential)             | Tries `AzureDeveloperCliCredential`, `AzureCliCredential`, `AzurePowerShellCredential`, and other credentials sequentially until one of them succeeds. Use this to have your application authenticate using developer tools, service principals, or managed identity based on what's available in the current environment without changing your code. |
```

}
```

#### Authenticating a User Account with Azure Developer CLI

> This example demonstrates authenticating the `SecretClient` from the [@azure/keyvault-secrets][secrets_client_library] client library using the `AzureDeveloperCliCredential` on a workstation with Azure Developer CLI installed and signed in.

Reviewer: ```suggestion
This example demonstrates authenticating the `SecretClient` from the [@azure/keyvault-secrets][secrets_client_library] client library using the `AzureDeveloperCliCredential` on a workstation with the Azure Developer CLI installed and signed in.
```

##### Configure Azure Developer CLI

Sign in using the [Azure Developer CLI](https://learn.microsoft.com/azure/developer/azure-developer-cli/reference?source=recommendations#azd-auth-login):

```bash
# As a user:
azd auth login

# As a service principal:
azd auth login --client-id <client-id> --client-secret <client-secret> --tenant-id <tenant-id>
```

##### Use Azure Developer CLI Credential

```ts
/**
 * Authenticate with Azure Developer CLI.
 */
function withAzureDeveloperCliCredential() {
  // As you can see in this example, the AzureDeveloperCliCredential does not take any parameters,
  // instead relying on the Azure Developer CLI authenticated user to authenticate.
  const credential = new AzureDeveloperCliCredential();

  const client = new SecretClient("https://key-vault-name.vault.azure.net", credential);
}
```

#### Authenticating in Azure with Workload Identity

> This example demonstrates authenticating the `SecretClient` from the [@azure/keyvault-secrets][secrets_client_library] using the `WorkloadIdentityCredential` in AKS environment.

Reviewer: ```suggestion
This example demonstrates authenticating the `SecretClient` from the [@azure/keyvault-secrets][secrets_client_library] using the `WorkloadIdentityCredential` in an AKS environment.
```
